### DISCUSSION TITLE
Battery power off by factor of 100

### DISCUSSION THREAD

**Discussion by viking2010** [hultenvp/solis-sensor#258] [General]
Hi

At about 2:40am this morning (14th Feb), the data from solis is showing that the battery power figure is out by a factor of 100. The actual solis cloud app is showing the right value, so just wanted to know if anyone else has had the same issue? Need to verify what the python test script shows, but will do so in the next hour and report back.

**Comment by sjoram** [comment]
Yes - same issue here

**Reply by viking2010** [reply]
Glad it's not just me then lol 😂. Will probably create a template sensor to change the value for the time being. Probably the quickest and easiest solution. Meanwhile, I'll log a ticket to solis!

**Reply by alienatedsec** [reply]
Yes the same for me. It's the main reason why I don't rely on the Solis Cloud anymore and don't use it as my only source of the Solar related information. I still use the integration, but I have made something custom, which is far more reliable and fit for purpose. https://github.com/fboundy/ha_solis_modbus/issues/4#issuecomment-1424738177

![image](https://user-images.githubusercontent.com/73167064/218677622-ba7246ef-a5fb-45a9-a019-db3b22195b19.png)

**Reply by viking2010** [reply]
That's an interesting project. I used to have the old style data logger and used the Solarman integration to get my data locally, but when I got remote access sorted from solis cloud, I had to change data loggers to the new version. I then had to come back to using this integration to get the data in to HA. For 2 months the Solis cloud data swapped my export and import around. They've no idea why! Support isn't always the best either which doesn't help.

**Reply by alienatedsec** [reply]
Expect some unannounced changes in the Solis Cloud as I can see changes to visual representation which makes no sense at all.

![image](https://user-images.githubusercontent.com/73167064/218709559-e3afe858-3f86-4136-96a4-8d39784f9ff1.png)

**Reply by kuttikutt** [reply]
yap  .. same here.

I guess they changed that on the API aswell...

Lets wait for a fix ...

**Reply by jmason** [reply]
I'm seeing the 100x scaling (but not any changes to the android app yet). very annoying.

**Reply by alienatedsec** [reply]
> I'm seeing the 100x scaling (but not any changes to the android app yet). very annoying.

That is on the web page, not an app.

**Comment by viking2010** [comment]
Seems all kind of screwed up! 😂

**Comment by LucidityCrash** [comment]
Looking at the api, and the code for the HA integration it would seem the API now returns kW instead of W for the units and is returning a number that is divided by 10 for the actual power figure. ie my battery is supplying 550 W at the moment, the api indicates -55kW which the code then shows as 55000 W.

The unit string that is returned in the API scales automatically so if the battery power was over 1000w it would change to 1 and change the units to kW.  The integration tries to take this into account ... as has been said it looks like internally they are multiplying the power number by 100 which results in what we are seing.

**Reply by hultenvp** [reply]
As lucidity writes the integration needs to keep the unit constant. HA does not (yet) accept variability in the unit of measurement.
They need to fix this on the API side.

**Reply by kuttikutt** [reply]
Hmm .. for now I added a template to Home Assistant: 

```
template:
  - sensor:
    - name: "BatterieleistungFuckUP"
      unit_of_measurement: "W"
      state: "{{ states('sensor.solis_battery_power')|float / 100 }}"

```
However: This can't be the solution - more a workaround...

So either the API-plugin switches cases and recalculate if it's W or kW by the factor, or we should start to open Tickets at GinLong/Solis.

What do you suggest?

**Reply by hultenvp** [reply]
Hi @kuttikutt: See also #259. I cannot predict the API's wrongful behavior, we know now how it misbehaves below 1kW, but above we have no clue. Introducing a workaround might cause more problems then it solves, it has in the past with other issues. I propose all who suffer from it submit a ticket, so they get some sense of urgency.

**Reply by kuttikutt** [reply]
Don't worry - my workaround is really only a workaround. I would not suggest anyone to implement this and happy to remove it ASAP :)
I actually read https://github.com/hultenvp/solis-sensor/issues/259 and this is as shitty fix as mine :) The API should deliver correct values.

Why I write these lines: I got little knowledge about the API and consider myself more as a user :) 

I would love to open a ticket at GinLong about his, but I don't know what to write. The info on this issue tracker is controversial. Some say it's a switch between W and kW (which would be 1000) .. but it's only 100. Others say that a negative value is interpreted as positive ...  

So what is exactly the issue? I'm not bad at coding - can you post some raw data what the API delivers?

**Comment by kuttikutt** [comment]
OK .. since I'm in close contact due to my ongoing Firmware updates, here is the reply I got:

```
 Hallo kutt,
​
It looks like there's a problem parsing the data for the API call, which is causing a deviation, there should be no such error on Soliscloud, right?
```

So they say: The problem is on the client side - lol

I already wrote a reply, that the data provided by the API changed and mixed up things on client side. (= stopped working properly w/o any changes locally)

**Comment by LucidityCrash** [comment]
LOL the issue is seen on the Web site and the app when you look at the
graph for battery power on the inverter details page ... though it does get
it right on the plant overview.

**Comment by cafedelmar01** [comment]
Looks like its fixed now!

**Comment by kuttikutt** [comment]
Guess this is partly my achievement, you can thank me later :) 

Had a chat with the Chinese support for the last 24h. Over 20 messages with explanations, screenshots etc. 
At first they didn't really understand what the issue was, later they blamed it on a wrong usage of the API, but finally they noticed that something must be wrong on their side.

But I have to admit - their support is quite fast. 

Seems to work like a charm on my HA as well now :) 

 ```
Hallo kutt,
​
The data of the API comes from our server, so the data of the API is exactly the same as the data of the platform;-）

My apologies...
​
Now the data seems normal.Please check.
```

**Reply by viking2010** [reply]
Nice work there! I can confirm that the data now seems to be working as it should.

**Reply by matthewbyrne** [reply]
Nice one. Looks good to me!
